share/completions/eopkg.fish

```diff
@@ -0,0 +1,153 @@
+#
+# Completions for the eopkg command
+#
+
+# Frequently used eopkg commands
+
+# Eopkg additional completion test
+function __fish_eopkg_package_ok -d 'Test if eopkg should have packages as completion candidate'
+    for i in (commandline -poc)
+        if contains -- $i upgrade up remove rm install it info check
+            return 0
+        end
+    end
+    return 1
+end
+
+function __fish_eopkg_component_ok -d 'Test if eopkg should have components as posible completion'
+    for i in (commandline -poc)
+        if contains -- $i -c --component
+            return 0
+        end
+    end
+    return 1
+end
+
+function __fish_eopkg_repo_ok -d 'Test if eopkg should have repositories as posible completion'
+    for i in (commandline -poc)
+        if contains -- $i remove-repo rr enable-repo er disable-repo dr
+            return 0
+        end
+    end
+    return 1
+end
+
+# Eopkg subcommand
+function __fish_eopkg_subcommand
+    set subcommand $argv[1]
+    set -e argv[1]
+    complete -f -c eopkg -n '__fish_use_subcommand' -a $subcommand $argv
+end
+
+function __fish_eopkg_subcommand_with_shortcut
+    set subcommand $argv[1]
+    set -e argv[1]
+    set shortcut $argv[1]
+    set -e argv[1]
+    complete -f -c eopkg -n '__fish_use_subcommand' -a $subcommand $argv
+    complete -f -c eopkg -n '__fish_use_subcommand' -a $shortcut $argv
+end
+
+function __fish_eopkg_using_subcommand -d 'Test if given subcommand is used'
+    for i in (commandline -poc)
+        if contains -- $i $argv
+            return 0
+        end
+    end
+    return 1
+end
+
+# Eopkg subcommand's option
+function __fish_eopkg_option
+    set subcommand $argv[1]
+    set -e argv[1]
+    complete -f -c eopkg -n "__fish_eopkg_using_subcommand $subcommand" $argv
+end
+
+function __fish_eopkg_option_with_shortcut
+    set subcommand $argv[1]
+    set -e argv[1]
+    set shortcut $argv[1]
+    set -e argv[1]
+    complete -f -c eopkg -n "__fish_eopkg_using_subcommand $subcommand" $argv
+    complete -f -c eopkg -n "__fish_eopkg_using_subcommand $shortcut" $argv
+end
+
+# Print additional completion
+function __fish_eopkg_print_components -d "Print list of components"
+    eopkg list-components -N | cut -d ' ' -f 1
+end
+
+function __fish_eopkg_print_repos -d "Print list of repositories"
+    eopkg list-repo -N | grep active | cut -d ' ' -f 1
+end
+
+# Setup additional completion
+complete -f -c eopkg -n '__fish_eopkg_package_ok; and not __fish_eopkg_component_ok' -a "(__fish_print_packages)" -d "package"
+complete -f -c eopkg -n '__fish_eopkg_component_ok' -a "(__fish_eopkg_print_components)" -d "component"
+complete -f -c eopkg -n '__fish_eopkg_repo_ok' -a "(__fish_eopkg_print_repos)" -d "repository"
+
+# Setup eopkg subcommand with shortcut
+__fish_eopkg_subcommand_with_shortcut upgrade up -d "Upgrades eopkg packages"
```

Comment:
Descriptions should be kept as short as possible - is the `eopkg` necessary here? That's the tool we're using, I wouldn't expect it to upgrade deb packages or rpms.